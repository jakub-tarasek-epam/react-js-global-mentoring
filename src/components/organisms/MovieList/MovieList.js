import React, { useEffect } from "react";
import { Col } from "react-bootstrap";
import MovieCard from "components/organisms/MovieCard/MovieCard";
import Navigation from "components/organisms/Navigation/Navigation";
import { RowMain, Wrapper } from "./MovieList.styles";
import { useSelector, useDispatch } from 'react-redux';
import { useParams } from "react-router";
import { fetchMovies } from "state/MoviesSlice";

const MovieList = () => {

  let { searchQuery } = useParams();
  const dispatch = useDispatch();

  const movies = useSelector((state) => {
    if(state.movies)
      return state.movies.movies
  });

  useEffect(() => {
    dispatch(fetchMovies(searchQuery));
  }, [searchQuery]);

  return (
    <Wrapper>
      <Navigation />
      <RowMain xs={1} md={3} lg={3} className="g-4">
        {movies && movies.map((movie) => (
          <Col key={movie.id}>
            <MovieCard movie={movie} />
          </Col>
        ))}
      </RowMain>
    </Wrapper>
  );
};

export default MovieList;
