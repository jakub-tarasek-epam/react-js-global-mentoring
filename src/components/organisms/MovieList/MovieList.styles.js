import styled from "styled-components";
import { Row } from "react-bootstrap";
import { colorsPallete } from "../../../theme/variables";

export const RowMain = styled(Row)`
    background-color: ${colorsPallete.almostDark};
    min-height: 50vh;
`;

export const Wrapper = styled.div`
    box-sizing: border-box;
`;

export const Nothing = styled.h2`
    color: ${colorsPallete.white};
    text-align: center;
`;