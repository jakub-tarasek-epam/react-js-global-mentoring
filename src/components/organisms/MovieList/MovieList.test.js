import React from 'react';
import routeData from 'react-router';
import renderer from 'react-test-renderer';

import MovieList from './MovieList';
import { useSelector } from 'react-redux';

jest.mock('react-redux', () => ({
    useSelector: jest.fn(),
    useDispatch: () => jest.fn(),
}));


describe('Movie list', () => {

    it('renders correctly movie list component', () => {
        jest.spyOn(routeData, 'useParams').mockReturnValue({ searchQuery: 'test' })
        useSelector.mockImplementation((selectorFn) => selectorFn({
            movies: {
                movies: [{
                    id: 1,
                    release_date: '1991-26-05',
                    title: 'title',
                    tagline: 'tagline'
                }]
            }
        }));
        const tree = renderer.create(<MovieList />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
