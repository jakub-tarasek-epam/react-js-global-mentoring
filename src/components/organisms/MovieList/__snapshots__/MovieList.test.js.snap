// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie list renders correctly movie list component 1`] = `
<div
  className="sc-egiyK iAXDMb"
>
  <div
    className="sc-pVTFL gNiyLU row"
  >
    <div
      className="sc-iqseJM gsLrwT hstack gap-3"
    >
      <div
        className="sc-jrQzAO itkHcF btn-group"
        role="group"
      >
        <input
          autoComplete="off"
          checked={true}
          className="btn-check"
          disabled={false}
          id="genreToggle-1"
          name="options"
          onChange={[Function]}
          type="radio"
          value="All"
        />
        <label
          className="sc-kDTinF fjZIUg btn btn-primary"
          href="link"
          htmlFor="genreToggle-1"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          ALL
        </label>
        <input
          autoComplete="off"
          checked={false}
          className="btn-check"
          disabled={false}
          id="genreToggle-2"
          name="options"
          onChange={[Function]}
          type="radio"
          value="Documentary"
        />
        <label
          className="sc-kDTinF fjZIUg btn btn-primary"
          htmlFor="genreToggle-2"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          DOCUMNETARY
        </label>
        <input
          autoComplete="off"
          checked={false}
          className="btn-check"
          disabled={false}
          id="genreToggle-3"
          name="options"
          onChange={[Function]}
          type="radio"
          value="Comedy"
        />
        <label
          className="sc-kDTinF fjZIUg btn btn-primary"
          htmlFor="genreToggle-3"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          COMEDY
        </label>
        <input
          autoComplete="off"
          checked={false}
          className="btn-check"
          disabled={false}
          id="genreToggle-4"
          name="options"
          onChange={[Function]}
          type="radio"
          value="Horror"
        />
        <label
          className="sc-kDTinF fjZIUg btn btn-primary"
          htmlFor="genreToggle-4"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          HORROR
        </label>
        <input
          autoComplete="off"
          checked={false}
          className="btn-check"
          disabled={false}
          id="genreToggle-5"
          name="options"
          onChange={[Function]}
          type="radio"
          value="Crime"
        />
        <label
          className="sc-kDTinF fjZIUg btn btn-primary"
          htmlFor="genreToggle-5"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          CRIME
        </label>
      </div>
      <div
        className="ms-auto"
      >
        SORT BY
      </div>
      <select
        onChange={[Function]}
      >
        <option
          value="release_date"
        >
          RELEASE DATE
        </option>
        <option
          value="vote_average"
        >
          RAITNG
        </option>
      </select>
    </div>
    <span>
      <strong>
        39
      </strong>
       movies found
    </span>
  </div>
  <div
    className="sc-crHmcD hjWyRA g-4 row row-cols-lg-3 row-cols-md-3 row-cols-1"
  >
    <div
      className="col"
    >
      <div
        className="sc-dkPtRN eZPrcG card"
      >
        <img
          className="card-img-top sc-furwcr hAkgKz"
          onClick={[Function]}
        />
        <div
          className="sc-jRQBWg eZJEUA forHover dropdown"
        >
          <span
            className="sc-bdvvtL ibuZAx"
            onClick={[Function]}
          >
            <div>
              <span />
              <span />
              <span />
            </div>
          </span>
        </div>
        <div
          className="sc-eCImPb xqahD card-body"
        >
          <div
            className="align-items-start hstack gap-2"
          >
            <div>
              <div
                className="sc-hKwDye hmqnzF card-title h5"
              >
                title
              </div>
              <div
                className="sc-gKclnd lnSLIm text-muted card-subtitle h6"
              />
            </div>
            <p
              className="sc-iCfMLu eLtYGZ align-items-center text-center ms-auto card-text"
            >
              1991-26-05
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
